controller.go

```diff
@@ -284,13 +284,17 @@ func syncPV(pv *PV) {
 			// NOTE: releasePV may either release the PV back into the pool or
 			// recycle it or do nothing (retain)
 
-			// HOWTO RELEASE A PV
-			pv.Status.Phase = Released
-			if err := CommitPVStatus(pv); err != nil {
-				// Status was not saved; we will fall back into the same
-				// condition in the next call to this method
-				return
+			// Make sure we don't overwrite previous failure of
+			// recycler/deleter.
+			if pv.Status.Phase != Failed {
```

Comment:
Also note I don't really base any our "important" decision on the phase, reclaim policy will be executed in all cases. Only the `Phase = Released` won't be set.